### PR TITLE
RHBA-415: After rename, designer is not marked as dirty more

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.jboss.org/browse/RHBA-415

Before:
before-1
before-2

After:
after-1
after-2

---

Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/275
- https://github.com/kiegroup/drools-wb/pull/842

### CODE REVIEW

**Comment by karreiro** [comment]
@mantis Precisely, Mike, that's the issue. The lock was not being released, and the editor was never marked as locked again.

There are two essential scenarios to cover:

- 1) The GDT Editor scenario:
    - Create a Guided Decision Table;
    - Rename the table;
    - Append some rows and add value to them;
    - (!) The editor was not being locked here.

- 1) Any other editor scenario (let's pick the DRL Editor):
    - Create a DRL file with the name "file";
    - Rename the DRL to "file1";
    - Make some changes in the file and save it;
    - Rename the DRL to "file2";
    - (!) The editor was not being locked and the title was not changing from "file1.drl - DRL" to "file2.drl - DRL".

File: uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/BaseEditorTest.java [hunk 1]
```diff
@@ -369,6 +381,134 @@ public void testMakeMenuBarWhenItDoesNotContainAllMenuItems() {
...
+        assertEquals(editorTitle, event.getTitleDecoration());
+    }
+
+    @Test
+    public void testInitVersionRecordManager() {
+
+        final ObservablePath path = mock(ObservablePath.class);
+        final PlaceRequest placeRequest = mock(PlaceRequest.class);
+        final String version = "version";
+        final Callback<VersionRecord> callback = (v) -> {
```
Comment (manstis) [inline uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/BaseEditorTest.java]: Might have been more readable to simply use a mock Callback?

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/LockManagerImpl.java [hunk 2]
```diff
@@ -273,21 +273,33 @@ private boolean isLockedByCurrentUser() {
         return lockInfo.isLocked() && lockInfo.lockedBy().equals(user.getIdentifier());
     }
 
-    private void updateLockInfo(@Observes LockInfo lockInfo) {
-        if (lockTarget != null && lockInfo.getFile().equals(lockTarget.getPath())) {
+    void updateLockInfo(final @Observes LockInfo lockInfo) {
+        if (getLockTarget() != null && lockInfo.getFile().toURI().equals(lockTarget.getPath().toURI())) {
```
Comment (manstis) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/LockManagerImpl.java]: You make a point of using ```toURI()``` on the paths here; but.....

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/LockManagerImpl.java [hunk 3]
```diff
@@ -309,6 +321,12 @@ void onSaveInProgress(@Observes SaveInProgressEvent evt) {
         }
     }
 
+    void onRenameInProgress(@Observes RenameInProgressEvent event) {
+        if (getLockTarget() != null && event.getPath().equals(lockTarget.getPath())) {
```
Comment (manstis) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/LockManagerImpl.java]: ... here you're OK using ```Path``` equality? (not URI, like your change above). Is this OK?

**Comment by karreiro** [comment]
@jomarko Good point. I'm not sure either. IMHO, the versions should be tracked right after every IO operation and not be lost.

Since it's not a regression, I think that this is not a blocker to this PR (but maybe we could raise a JIRA for that), see:

Before:
2018-04-05 6 46 45 pm

After:
2018-04-05 6 07 54 pm

File: uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/BaseEditorTest.java (hunk 1 above)
Comment (karreiro) [reply uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/test/java/org/uberfire/ext/editor/commons/client/BaseEditorTest.java]: Good catch. Fixed.

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/LockManagerImpl.java (hunk 2 above)
Comment (karreiro) [reply uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/LockManagerImpl.java]: @manstis Here we can have a `PathImpl` being compared to an `ObservablePath` (that's what we're avoiding), and......

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/LockManagerImpl.java (hunk 3 above)
Comment (karreiro) [reply uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/mvp/LockManagerImpl.java]: ...here, we _always_ have an `ObservablePath` being compared to another `ObservablePath`.

I added a commentary to make this less confusing.

Thanks for the comment :-)

**Pull request merged**